src/vs/workbench/contrib/terminal/browser/terminalGroupService.ts

```diff
@@ -181,12 +181,7 @@ export class TerminalGroupService extends Disposable implements ITerminalGroupSe
 				}
 				// Do not await the focus as setVisible must be called immediately to ensure
 				// something else didn't steal focus
-				instance.focusWhenReady(true);
-				// HACK: as a workaround for https://github.com/microsoft/vscode/issues/134692,
-				// this will trigger a forced refresh of the viewport to sync the viewport and scroll bar.
-				// This can likely be removed after https://github.com/xtermjs/xterm.js/issues/291 is
-				// fixed upstream.
-				instance.setVisible(true);
+				await instance.focusWhenReady(true);
```

Comment:
I think it's save to remove this hack after the refactor in https://github.com/microsoft/vscode/pull/149621, I can't reproduce it anymore. Also tested https://github.com/microsoft/vscode/issues/160914 and is working fine too